### ISSUE TITLE
Unable to choose account from drop down menu

### ISSUE THREAD

**Issue by jeffreycoe** [spinnaker/spinnaker#5672]
### Issue Summary:
When attempting to configure a pipeline stage, we're unable to choose an account from the accounts drop down list.

### Cloud Provider(s):
AWS
Kubernetes

### Environment:
Any

### Feature Area:
Pipeline, UI

### Description:
When attempting to configure a pipeline stage, we're unable to choose an account from the accounts drop down list.

This issue does happen with not just AWS account drop down menus but with account drop down menus for Kubernetes stages, too.

### Steps to Reproduce:
Our spinnaker deployment is currently configured with 2 AWS accounts and 1 Kubernetes account. 

For this specific example, we create a new application, add the aws and kubernetes providers to the application as we have stages which will use both providers, and then create a new pipeline.  

Within the new pipeline, we add a stage using the Deploy (CloudFormation Stack) stage type. Within this stage, we attempt to select one of the accounts from the account drop down menu, but it never selects the account. We're unable to configure the pipeline at all.

We have not found any errors in the container logs nor do we see any errors in the console in the browser.

See the below screenshot which demonstrates this issue:
![spinnaker-bug](https://user-images.githubusercontent.com/13616837/79159880-e350fc00-7da6-11ea-86af-97daf39c5889.gif)

### Additional Details:
N/A

**Comment by k1nger** [comment]
I've noticed that the AccountSelectField in the ts code is being reset at some point in the $timeout method. Seems like a front end issue? While it's setting the timeout, it has the right variable. When it gets into the timeout though the variable is back to "select"...

**Comment by magruder85** [comment]
Also, seeing these following errors in clouddriver:

"AmazonCloudFormationException: Rate exceeded"

**Comment by maggieneterval** [comment]
Hi @jeffreycoe, thanks for taking the time to open this issue. Just to clarify, are you only experiencing this issue with the Deploy CloudFormation Stack stage? What version of Spinnaker are you running?

**Comment by magruder85** [comment]
Hi @maggieneterval,

We upgraded to 1.20.0 on Monday and the selector started working for us. I have been able to deploy CloudFormation stages now. We had tested our spinnaker configuration on a scrum spinnaker instance and we had no issues with the dropdown selector there, so it seemed very specific to our deployment in Kubernetes. We were going to run a hal deploy clean to re-deploy a clean configuration, but the upgrade seems to have helped us out.

**Comment by maggieneterval** [comment]
Hi @magruder85, thanks for the update! Glad to hear that things are working for you now; going to close this issue but please feel free to re-open if you encounter this problem again.

Thanks!

**Comment by sinn1** [comment]
> I've noticed that the AccountSelectField in the ts code is being reset at some point in the $timeout method. Seems like a front end issue? While it's setting the timeout, it has the right variable. When it gets into the timeout though the variable is back to "select"...
> 
> More specifically in accountSelecetField.component.ts
> 
> ```
> this.handleSelectChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
>         $timeout(() => {
>           this.currentValue = this.component[this.field] = event.target.value;
>           this.onChange && this.onChange();
>         });
>       };
> ```
> 
> eveent.target.value has the right value before the $timeout is executed, but when it's executed it's ste back to default.
> 
> I also found this commeent in the AccountSelecetInput.tsx
> 
> ```
> // When this select is used in Angular, the event is accessed in a $timeout, and React will have
> // re-rendered the input, setting its value (the event.target.value) back to the previous value
> // This can go away once we're out of Angular land.
> ```
> 
> Which seems to be the exact issue we are running into.

I'm experiencing this exact issue when creating a Server Group for AppEngine. Is there a workaround? I've had to add a breakpoint to the code and manually update 'event.target.value' to the correct value.

(Let me know if you'd prefer me to open a separate ticket).
